src/middlewared/middlewared/plugins/multipath.py

```diff
@@ -56,86 +49,37 @@ def query(self, filters, options):
+                provref = i.find('./provider').attrib['ref']
+                prov = doc.findall(f'.//provider[@id="{provref}"]')[0]
+                da_name = prov.find('./name').text
+                try:
+                    lun_id = prov.find('./config/lunid').text
+                except Exception:
+                    lun_id = ''
+
                 children.append({
-                    "type": "consumer",
-                    "name": cn.devname,
-                    "status": cn.status,
-                    "lun_id": cn.lunid,
+                    'type': 'consumer',
+                    'name': da_name,
+                    'status': consumer_status,
+                    'lun_id': lun_id,
                 })
 
-            data = {
-                "type": "root",
-                "name": mp.devname,
-                "status": mp.status,
-                "children": children,
-            }
-            items.append(data)
-
-        return filter_list(items, filters=filters or [], options=options or {})
-
-    def __get_multipaths(self):
-        doc = etree.fromstring(sysctl.filter("kern.geom.confxml")[0].value)
-        return [
-            Multipath(doc=doc, xmlnode=geom)
-            for geom in doc.xpath("//class[name = 'MULTIPATH']/geom")
-        ]
-
-
-class Multipath(object):
-    """
-    Class representing a GEOM_MULTIPATH
-    """
-
-    @property
-    def status(self):
-        return getattr(self, "_status", "Unknown")
-
-    @status.setter
-    def status(self, value):
-        self._status = value
-
-    @property
-    def devices(self):
-        devs = []
-        for consumer in self.consumers:
-            devs.append(consumer.devname)
-        return devs
-
-    def __init__(self, doc, xmlnode):
-        self.name = xmlnode.xpath("./name")[0].text
-        self.devname = f"multipath/{self.name}"
-        self._status = xmlnode.xpath("./config/State")[0].text
-        self.consumers = []
-        for consumer in xmlnode.xpath("./consumer"):
-            status = consumer.xpath("./config/State")[0].text
-            provref = consumer.xpath("./provider/@ref")[0]
-            prov = doc.xpath(f"//provider[@id = '{provref}']")[0]
-            self.consumers.append(Consumer(status, prov))
-
-        self.__xml = xmlnode
-        self.__doc = doc
-
-    def __repr__(self):
-        return f"<Multipath:{self.name} [{','.join(self.devices)}]>"
-
-
-class Consumer(object):
-
-    def __init__(self, status, xmlnode):
-        self.status = status
-        self.devname = xmlnode.xpath("./name")[0].text
-        try:
-            self.lunid = xmlnode.xpath("./config/lunid")[0].text
-        except Exception:
-            self.lunid = ""
-        self.__xml = xmlnode
+            result.append({
+                'type': 'root',
+                'name': 'multipath/' + g.find("./name").text,
+                'status': g.find("./config/State").text,
+                'children': children,
+            }),
+        else:
```

Comment:
When would we never reach this block ?

Reply:
`else:` is reached if you have a system with no multipath disks.

Reply:
I am not sure i understand why we need the `else` block ? Actions performed in `else` need to be performed in either cases. This could even potentially introduce a bug if later a break is added in the loop.

Reply:
It's old debugging code I was  using on live system. I've removed it.